### PR TITLE
Fix Shape.hitsShape, plus other miscellaneous new tests in issues.py

### PR DESCRIPTION
**Pull request by schmave**

### CODE REVIEW

File: cmu_graphics/utils.py [hunk 1]
```diff
@@ -113,6 +115,8 @@ def isGroup(shape):
 
 def getChildShapes(shape):
     result = []
+    if hasattr(shape, '_shape'):
```
Comment (schmave) [inline cmu_graphics/utils.py:118]: @austin-schick When you passed a group as an argument to Shape.hitsShape(), the child shapes of that group were not converted to shape_logic.Shape. Is this the right way to fix that?

File: cmu_graphics/utils.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ def polygonContainsPoint(pts, px, py):
         q1y = q1[1]
         q2x = q2[0]
         q2y = q2[1]
+        if distanceToLineSegment2(px, py, q1x, q1y, q2x, q2y) < 0.0002:
```
Comment (austin-schick) [inline cmu_graphics/utils.py:62]: What's going on on this line?

File: cmu_graphics/utils.py (hunk 1 above)
Comment (austin-schick) [reply cmu_graphics/utils.py:118]: This looks like a good solution. How is this working in `cmu-graphics.js` right now? I just took a look and didn't see a conversion happening

File: cmu_graphics/utils.py (hunk 2 above)
Comment (schmave) [reply cmu_graphics/utils.py:62]: This is a port of https://github.com/cmu-cs-academy/cs-academy/pull/2708

File: cmu_graphics/utils.py (hunk 1 above)
Comment (schmave) [reply cmu_graphics/utils.py:118]: It's handled by `jsApply`. Maybe the difference is that JS Group objects only hold references to JS Shapes, while in desktop CMU Graphics SLI.Group objects can refer to cmu_graphics.Shapes?

**Pull request merged**